tutorials/tmva/TMVA_CNN_Classification.py

```diff
@@ -0,0 +1,370 @@
+# TMVA Classification Example Using a Convolutional Neural Network
+
+# This is an example of using a CNN in TMVA. We do classification using a toy image data set that is generated when running the example macro.
+
+# Helper function to create input images data we create a signal and background 2D histograms from 2d gaussians with a location (means in X and Y)  different for each event The difference between signal and background is in the gaussian width. The width for the background gaussian is slightly larger than the signal width by few % values
+
+# First you need to run TMVA_CNN_Classification.C to generate images_data_16x16.root.
+
+import ROOT
+from ROOT import TMVA 
+import os
+from array import array
+import tensorflow
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.optimizers import Adam
+
+from tensorflow.keras.layers import Input, Dense, Dropout, Flatten, Conv2D, MaxPooling2D, Reshape, BatchNormalization
+
+# Setting up TMVA
+
+ROOT.TMVA.Tools.Instance()
+# TMVA requires initialization the PyMVA to utilize PyTorch. PyMVA is the interface for third-party MVA tools based on Python. It is created to make powerful external libraries easily accessible with a direct integration into the TMVA workflow. All PyMVA methods provide the same plug-and-play mechanisms as the TMVA methods. Because the base method of PyMVA is inherited from the TMVA base method, all options of internal TMVA methods apply for PyMVA methods as well.
+# For PYMVA methods
+TMVA.PyMethodBase.PyInitialize()
+
+#  Create an Output File and Declare Factory
+# 
+# Create the Factory class. Later you can choose the methods whose performance you'd like to investigate.
+# 
+# The factory is the major TMVA object you have to interact with. Here is the list of parameters you need to pass
+# 
+# - The first argument is the base of the name of all the output weightfiles in the directory weight/ that will be created with the method parameters
+# 
+# - The second argument is the output file for the training results
+# 
+# - The third argument is a string option defining some general configuration for the TMVA session. For example all TMVA output can be suppressed by removing the "!" (not) in front of the "Silent" argument in the option string
+# 
+
+outputFile = ROOT.TFile.Open("CNN_ClassificationOutput.root", "RECREATE")
+
+factory = ROOT.TMVA.Factory("TMVA_CNN_Classification", outputFile,
+                      "!V:ROC:!Silent:Color:!DrawProgressBar:AnalysisType=Classification" )
+
+
+# Define the Options and number of threads
+
+
+opt=[1,1,1,1,1]
```

Comment:
We could use a more Pythonic way to pass the option to the tutorial. Maybe some command line argument ?